src/main.rs

```diff
@@ -0,0 +1,27 @@
+mod config;
+mod error;
+mod event;
+mod source;
+
+use config::NatsConfig;
+
+use fluvio::{RecordKey, TopicProducer};
+use fluvio_connector_common::{
+    connector,
+    tracing::{debug, trace},
+    Result, Source,
+};
+use futures::StreamExt;
+use source::NatsSource;
+
+#[connector(source)]
+async fn start(config: NatsConfig, producer: TopicProducer) -> Result<()> {
+    debug!(?config);
+    let source = NatsSource::new(&config)?;
+    let mut stream = source.connect(None).await?;
```

Comment:
Depending on the long run behavior of Nats/fluvio, it may be interesting to have a larger loop around the connect .. next to implement a longer term re-connect strategy

Reply:
Yeah we just crashed the connector and there wasn't really good feedback loop to cluster to handle failing connectors.. it would be good to have some kind of strategy e.g. also observability of these failures